### PR TITLE
3.x Upgrade Slurm Accounting template to Serverless V2

### PR DESCRIPTION
**Pull request by davprat**
### Description of changes
* Aurora Serverless V1 is not supported in eu-north-1
* This updates the template to use V2 for the Slurm accounting cluster database.
* This change also adds a parameter to allow a user to specify the name of the database cluster - using the generated Cfn name could result in the hostname being longer than the 63 character limit allowed by MySQL.

### Tests
* Ran accounting integration tests in us-east-1, eu-north-1, and ap-south-1.

### Checklist
- [x] Make sure you are pointing to **the right branch** and add a label in the PR title (i.e. **2.x** vs **3.x**)
- [x] Check all commits' messages are clear, describing what and why vs how.
- [x] Make sure **to have added unit tests or integration tests** to cover the new/modified code.

Please review the [guidelines for contributing](../CONTRIBUTING.md) and [Pull Request Instructions](https://github.com/aws/aws-parallelcluster/wiki/Git-Pull-Request-Instructions).

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cloudformation/database/serverless-database.yaml [hunk 1]
```diff
@@ -7,6 +7,7 @@ Metadata:
       - Label:
           default: "Database Cluster Configuration"
         Parameters:
+          - ClusterName
```
Comment (jdeamicis) [inline cloudformation/database/serverless-database.yaml:10]: Was this necessary only for serverless v2? Or is this a general improvement to the template?

File: cloudformation/database/serverless-database.yaml [hunk 2]
```diff
@@ -237,106 +235,127 @@ Resources:
...
         - Key: 'parallelcluster:usecase'
           Value: 'slurm accounting'
-  ServerlessDatabaseCluster:
+  AccountingCluster:
     Type: 'AWS::RDS::DBCluster'
     Properties:
-      Engine: aurora-mysql
+      DBClusterIdentifier: !Ref ClusterName
+      Engine: "aurora-mysql"
+      EngineVersion: "8.0.mysql_aurora.3.02.1"
```
Comment (jdeamicis) [inline cloudformation/database/serverless-database.yaml:268]: (minor) is there a way to pick the latest one by any chance?

File: cloudformation/database/serverless-database.yaml [hunk 3]
```diff
@@ -237,106 +235,127 @@ Resources:
...
+    Type: 'AWS::RDS::DBInstance'
+    Properties:
+      DBInstanceClass: 'db.serverless'
+      DBClusterIdentifier: !Ref AccountingCluster
+      DBInstanceIdentifier: !Sub '${ClusterName}-instance-1'
+      Engine: "aurora-mysql"
+      PubliclyAccessible: false
+    UpdateReplacePolicy: Delete
+    DeletionPolicy: Delete
+  AccountingClusterInstance2:
```
Comment (jdeamicis) [inline cloudformation/database/serverless-database.yaml:298]: Oh, why do we need two instances? Is it for multi-az and reader/writer?

**Review by jdeamicis — APPROVED** [review]

File: cloudformation/database/serverless-database.yaml (hunk 1 above)
Comment (davprat) [reply cloudformation/database/serverless-database.yaml:10]: General improvement - default, generated names could exceed the 63 character limit.

File: cloudformation/database/serverless-database.yaml (hunk 2 above)
Comment (davprat) [reply cloudformation/database/serverless-database.yaml:268]: I don't know? I can look into it - maybe a parameter type exists for selecting this value or maybe there is a built-in function (though I haven't come across it)

File: cloudformation/database/serverless-database.yaml (hunk 3 above)
Comment (davprat) [reply cloudformation/database/serverless-database.yaml:298]: Strictly, I don't think we do (my original testing only had one), but the console creates 2 by default so I was trying to mirror that.

**Pull request merged**